testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergProcedureCalls.java

```diff
@@ -59,6 +60,134 @@ public void testMigrateHiveTable()
+                        3, "2021-01-01 10:11:12.123",
+                        "2021-01-01 10:11:12.123000 UTC",
+                        "2021-01-01 10:11:12.123"),
+                new TimestampTestCase("MICROSECONDS",
+                        6, "2021-01-01 10:11:12.123456",
+                        "2021-01-01 10:11:12.123456 UTC",
+                        "2021-01-01 10:11:12.123456"),
+                new TimestampTestCase("NANOSECONDS",
+                        9, "2021-01-01 10:11:12.123456789",
+                        "2021-01-01 10:11:12.123457 UTC",
+                        "2021-01-01 10:11:12.123456")
+        ).forEach(testCase ->
+                testMigrateTimestampHiveTable(
+                        testCase.precisionName,
+                        testCase.precision,
+                        testCase.inputValue,
+                        testCase.expectedValueInTrino,
+                        testCase.expectedValueInSpark,
+                        "PARQUET"));
+    }
+
+    private record TimestampTestCase(String precisionName, int precision, String inputValue, String expectedValueInTrino, String expectedValueInSpark) {}
+
+    private void testMigrateTimestampHiveTable(String precisionName, int precision, String inputValue, String expectedValueInTrino, String expectedValueInSpark, String format)
+    {
+        String tableName = "test_migrate_timestampPar_" + randomNameSuffix();
+        String hiveTableName = "hive.default." + tableName;
+        String icebergTableName = "iceberg.default." + tableName;
+        String sparkTableName = "iceberg_test.default." + tableName;
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS default." + tableName);
+        onTrino().executeQuery("SET SESSION hive.timestamp_precision = '" + precisionName + "'");
+        onTrino().executeQuery("CREATE TABLE " + hiveTableName + " (x timestamp(" + precision + ")) WITH (format='" + format + "')");
+        onTrino().executeQuery("INSERT INTO " + hiveTableName + " VALUES timestamp '" + inputValue + "'");
+
+        assertThat(onHive().executeQuery("SELECT CAST(x AS string) FROM default." + tableName))
+                .containsOnly(row(inputValue));
+        assertThat(onTrino().executeQuery("SELECT CAST(x AS varchar) FROM " + hiveTableName))
+                .containsOnly(row(inputValue));
+
+        onTrino().executeQuery("CALL iceberg.system.migrate('default', '" + tableName + "')");
+
+        assertThat(onTrino().executeQuery("SELECT CAST(x AS varchar) FROM " + icebergTableName))
+                .containsOnly(row(expectedValueInTrino));
+        assertThat(onSpark().executeQuery("SELECT CAST(x AS string) FROM " + sparkTableName))
+                .containsOnly(row(expectedValueInSpark));
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + icebergTableName);
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testMigrateTimestampHiveTableInComplexType()
```

Comment:
I think we should move this test to `io.trino.plugin.iceberg.TestIcebergMigrateProcedure`. We don't need comparison with Hive and Spark for this one.